Student/StudentUITests/Modules/ModulesTests.swift

```diff
@@ -120,3 +121,73 @@ class ModulesTests: CoreUITestCase {
         ModulesDetail.moduleItem(index: 0).waitToExist()
     }
 }
+
+class MockedModulesTests: StudentUITestCase {
+    override var experimentalFeatures: [ExperimentalFeature] {
+        return [.newPageDetails]
+    }
+
+    override func setUp() {
+        super.setUp()
+        mockBaseRequests()
+    }
+
+    func testUnlockModuleItemWithPrerequisiteModule() {
+        let modules: [APIModule] = [
+            .make(id: "1", name: "Module 1", position: 1, prerequisite_module_ids: [], state: .unlocked),
+            .make(id: "2", name: "Module 2", position: 2, prerequisite_module_ids: ["1"], state: .locked),
+        ]
+        mockData(GetModulesRequest(courseID: "1", include: [.items], perPage: 99), value: modules)
+        mockData(GetModuleItemsRequest(courseID: "1", moduleID: "1", include: [.content_details, .mastery_paths], perPage: 99), value: [
+            .make(
+                id: "1",
+                module_id: "1",
+                title: "Page 1",
+                content: .page("page-1"),
+                url: URL(string: "https://canvas.instructure.com/api/v1/courses/1/pages/page-1")!,
+                completion_requirement: .make(type: .must_view, completed: false)
+            ),
+        ])
+        mockData(GetModuleItemsRequest(courseID: "1", moduleID: "2", include: [.content_details, .mastery_paths], perPage: 99), value: [
+            .make(
+                id: "2",
+                module_id: "2",
+                title: "Page 2",
+                content: .page("page-2"),
+                url: URL(string: "https://canvas.instructure.com/api/v1/courses/1/pages/page-2")!,
+                content_details: .make(locked_for_user: true, lock_explanation: "This item is part of a prereq module")
+            ),
+        ])
+        mockData(
+            GetModuleItemSequenceRequest(courseID: "1", assetType: .moduleItem, assetID: "1"),
+            value: APIModuleItemSequence(
+                items: [.init(prev: nil, current: .make(id: "1", module_id: "1"), next: nil)],
+                modules: []
+            )
+        )
+        mockData(GetPageRequest(context: ContextModel(.course, id: "1"), url: "page-1"), value: .make())
+        mockData(PostMarkModuleItemRead(courseID: "1", moduleID: "1", moduleItemID: "1"))
+        show("courses/1/modules")
+        XCTAssertEqual(Modules.module(index: 0).label(), "Module 1")
+        XCTAssertEqual(Modules.module(index: 1).label(), "Module 2. Status: Locked")
+        Modules.module(index: 1).tap()
+        app.find(labelContaining: "PREREQUISITE MODULES").waitToExist()
+        XCTAssertEqual(app.find(id: "module_cell_0_0").label(), "Module 1")
+        XCTAssertEqual(ModulesDetail.moduleItem(index: 0).label(), "Page 2. Type: Page. Status: Locked")
+        ModulesDetail.module(index: 0).tap()
+        ModulesDetail.moduleItem(index: 0).tap()
```

Comment:
test doesn't execute this line correctly for me. Race condition with loading maybe?

Reply:
I'll check. Maybe I broke something with a refactor or a merge.

Reply:
@cobbal the problem was a deadlock on the main thread. Should be good now.